Save current config to .yo-rc.json

pitrew opened the pull request:
This will save user answers to local storage as well. Then next time you execute yo it will first try to find global storage then local and override with local prompts.

This is very useful when I produce multiple projects (components) using the same generator. Then next time I want to update I wont be answering same questions again.

SBoudrias commented:
Sorry for the delayed review, I think I'll have some time to go over it tonight.

SBoudrias commented:
This look good to me, mind adding unit tests?

eddiemonge commented:
ping @pitrew

pitrew commented:
Hey, I'll add them

On Oct 24, 2016 5:57 PM, "Eddie Monge" <notifications@github.com> wrote:

> ping @pitrew <https://github.com/pitrew>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/yeoman/generator/pull/963#issuecomment-255782750>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAs3JVNwbC4UqpxeEpsfyzJSl3kHHAA0ks5q3NVogaJpZM4KV0Vd>
> .
>

pitrew commented:
Hi I've added tests and fixed raw list. In tests it was getting array of strings (and sepeartor) but the function was doing `_.map(ans, 'value')` whlist the input was a string. This was causing not saving defaults for lists when specified as arrays only

SBoudrias commented on lib/util/prompt-suggestion.js:
```diff
@@ -54,15 +54,21 @@ var getListDefault = function (question, defaultValue) {
  *
  * @param  {Object}       question Inquirer prompt item
  * @param  {String|Array} answer   The inquirer answer
+ * @param  {Boolean}      storeAll Should store default values
```
I wonder if there's really any case where we don't want to store the default value. Given that we only update the default value of the question array, it wouldn't ever be a problem to store the default one right?

pitrew replied:
@SBoudrias The original behaviour was not to store default values. I didn't want to break this. So storeAll set to false will reflect the original workflow. What do you suggest?

SBoudrias commented on test/prompt-suggestion.js:
```diff
@@ -293,6 +331,44 @@ describe('PromptSuggestion', function () {
+        respuesta: 'foo'
+      };
+
+      this.store.delete('promptValues');
```
This should be done as part of the teardown.

pitrew replied:
This is rather a setup. I moved it to beforeEach

pitrew commented:
ping @eddiemonge, @SBoudrias

SBoudrias commented:
Okay, sorry for the time it took me to review this. I had to sit down and wrap my head around the impact and make sure it wouldn't be problematic. So yeah, that all sounds good. Let's merge :)

Thanks for taking the time.
